Penteract/Assets/Scripts/Player.cpp

```diff
@@ -225,24 +225,29 @@ bool Player::GetInputBool(InputActions action, bool useGamepad) {
 
 }
 
-float2 Player::GetInputFloat2(InputActions action, bool useGamepad) const {
-	float2 result = float2(0, 0);
+float2 Player::GetInputFloat2(InputActions action, bool useGamepad) {
+	result = float2(0, 0);
+
 	switch (action) {
 	case InputActions::MOVEMENT:
 		if (useGamepad && Input::IsGamepadConnected(0)) {
 			result.x = Input::GetControllerAxisValue(Input::SDL_CONTROLLER_AXIS_LEFTX, 0);
 			result.y = Input::GetControllerAxisValue(Input::SDL_CONTROLLER_AXIS_LEFTY, 0);
 		} else {
 			if (Input::GetKeyCodeDown(Input::KEYCODE::KEY_D) || Input::GetKeyCodeRepeat(Input::KEYCODE::KEY_D)) {
+				Debug::Log("1");
```

Comment:
Remove logs

Reply:
You can change de angle and the offset in the inspector

Reply:
i put on the .h the optim value but can be modify on the inspector

Reply:
**You've just put NAN on Level2. careful with that.** 

Also, laser should properly follow Onimaru's Weapon? 

![image](https://user-images.githubusercontent.com/27434068/135756598-2508a3e1-9602-4298-968d-51f1766a8b54.png)